### PR TITLE
Cleanup auxilia script

### PR DESCRIPTION
**Pull request by rpelisse**
- Delete extra scripts
- Rename main script as auxilia
- Added validation and sanity check (2 remains to be implemented)
- Ensure scripts stops if PR metadata cannot be retrieved
- Fixed up syntax to follow shell best practises (no backquote for command substitution)

### CODE REVIEW

**Comment by rpelisse** [comment]
@wolfc I guess those changes should be reviewed by anyone who has done a release and had to use the merge-pr script.

**Comment by spyrkob** [comment]
What's the reason for name change? 'merge-pr' was nice and descriptive, "does what it says on the tin"  :)

**Comment by rpelisse** [comment]
I prefer to have name matching the repo so that if you have several scripts, there is no question what you should launch. It's also easier in conversation to refer to auxilia than merge-pr. That's being I'm fine with renaming the script merge-pr and simply add an symlink auxilia to it for instance.

**Comment by spyrkob** [comment]
> I prefer to have name matching the repo so that if you have several scripts, there is no question what you should launch. It's also easier in conversation to refer to auxilia than merge-pr. That's being I'm fine with renaming the script merge-pr and simply add an symlink auxilia to it for instance.

I can see your point if the repo contains one main script/program with multiple sub-scripts. In this case I think the repo either is supposed to contain just this one script (in which case I'd argue that repo should be renamed), or a collection of utility scripts (in which case one entry point scripts doesn't make much sense).

Either way, the code changes look good. If everybody else is OK with the name change, I'm good too. Just a bit of Friday rant ;)

**Comment by TomasHofman** [comment]
I'm with @spyrkob over the renaming. It's gonna confuse me. One usually needs to use this once or twice a year, and I'm gonna forget everything during that time. Descriptive file name solves a lot.

**Comment by rstancel** [comment]
I also vote for not renaming. auxiliary one is not descriptive at all.

**Comment by rpelisse** [comment]
@spyrkob @TomasHofman @rstancel @baranowb @istudens @ivassile @xstefank @wolfc I've did a few more tweaks to my cleanup of the auxilia script (and I've renamed it back to pr-merge). Let me know if you see any problem with the changes, otherwise I'll merge them before the end of the week.

File: pr-merge [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+if [ "${?}" -ne 0 ]; then
+  echo "Failed."
+  exit 2
+fi
+echo 'Done.'
+set +e
+
+readonly MSG="Merge pull request #${PR_ID} from $(echo ${PULL} | jq -r .head.label) $(echo $PULL | jq -r .title)"
+echo "Git commit message will be: ${MSG}"
+#echo -n "Continue..." && read _
```
Comment (xstefank) [inline pr-merge]: intentionally commented?

Comment (rpelisse) [reply pr-merge]: Oh good catch, I actually wanted to wrap this into an option so one could do "pr-merge -y" (for instance) to skip this. Let me uncomment it, i'll do some fancy options handling in a separate PR.

Comment (rpelisse) [reply pr-merge]: Done.

File: pr-merge [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+fi
+
+if [ -z "${PR_ID}" ]; then
+  echo "No PR provided."
+  usage
+  exit 3
+fi
+
+echo "Merging ${PR_ID} from ${REMOTE} onto ${BRANCH}"
+readonly FETCH_URL=$(git remote -v | grep "${REMOTE}" | grep 'fetch' | cut -f2 | cut -d" " -f1)
```
Comment (gaol) [inline pr-merge]: Maybe `git remote -v | egrep "^${REMOTE}[ ]*"` will be more accurate for the git remote?

It also would be good to check whether the `FETCH_URL` is valid to decide whether proceed or not IMO. :)

File: pr-merge [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+readonly OWNER=$(echo "${FETCH_URL}" | cut -d: -f2 | cut -d/ -f1)
+readonly REPO=$(echo "${FETCH_URL}" | cut -d: -f2 | cut -d/ -f2 | cut -f1 -d. )
+readonly PR_GITHUB_API_URL="https://api.github.com/repos/${OWNER}/${REPO}/pulls/${PR_ID}"
+set -e
+echo -n "Retrieving PR information from ${PR_GITHUB_API_URL}..."
```
Comment (gaol) [inline pr-merge]: do you mean `echo -e` ? :)

Comment (rpelisse) [reply pr-merge]: Nop, I do mean -n here:
`-n     do not output the trailing newline`

It will give this kind of output:

`Retrieving PR information from ${PR_GITHUB_API_URL}...Done.`
or

`Retrieving PR information from ${PR_GITHUB_API_URL}...Failed.`

(so  one line for the all operation)

I'm not sure what -e will bring us:

`-e     enable interpretation of backslash escapes`

File: pr-merge (hunk 2 above)
Comment (rpelisse) [reply pr-merge]: In fact, grep can handle this kind of regexp, but yes, it's a good point :) 

And you are right, we could do a curl -I ${FETCH_URL} to check that URL is valid. I'll add that right away.

Comment (rpelisse) [reply pr-merge]: Darn, I did not though about that, but the FETCH_URL is not always using http (git@github.com:orga/project.git), so CURL won't work... I'll just add a test to check the value is not empty. I don't think we can more than that here.
